src/coreclr/vm/eventtrace.cpp

```diff
@@ -6655,6 +6655,29 @@ VOID ETW::MethodLog::SendMethodEvent(MethodDesc *pMethodDesc, DWORD dwEventOptio
     // by a TADDR (i.e., w/out thumb bit set on ARM)
     EECodeInfo codeInfo(start);
 
+    // We can race with the JIT publishing GCInfo, if it is not published we can skip this method since
+    // it is not in use yet.
+#ifdef TARGET_X86
+    hdrInfo  gcInfo;
+    DecodeGCHdrInfo(codeInfo.GetGCInfoToken(),
```

Comment:
Should this rather check for NULL GetGCInfoToken? I do not think we should be attempting to decode the GC info that has not been published.

Reply:
Or maybe this change is not needed with the new approach?